tau.h

```diff
@@ -273,8 +273,14 @@ class TauCorrProvider : public CorrectionsBase<TauCorrProvider> {
                                            ? scale : UncScale::Central;
             const UncSource tau_ele_source = tau_ele_scale == UncScale::Central ? UncSource::Central : source ;
             const std::string& scale_str = getScaleStr(tau_ele_source, tau_ele_scale, year_);
-            const auto sf = tau_vs_e_->evaluate({Tau_p4.eta(), Tau_genMatch, wpVSe, scale_str});
-            return sf;
+            if (year_.starts_with("202"s) == 1){
```

Comment:
Save the boolean in the constructor. Don't perform unnecessary string operations for each event.